src/platforms/esp32/components/avm_builtins/gpio_driver.c

```diff
@@ -69,10 +68,16 @@ static Context *gpio_driver_create_port(GlobalContext *global, term opts);
 #endif
 
 #ifdef CONFIG_AVM_ENABLE_GPIO_PORT_DRIVER
-static const char *const gpio_atom = "\x4" "gpio";
-static const char *const gpio_driver_atom = "\xB" "gpio_driver";
+static const char *const gpio_atom = ATOM_STR("\x4", "gpio");
+static const char *const gpio_driver_atom = ATOM_STR("\xB", "gpio_driver");
+static const char *const gpio_interrupt_atom = ATOM_STR("\xE", "gpio_interrupt");
 #endif
 
+static const char *const high_atom = ATOM_STR("\x4", "high");
+static const char *const low_atom_str = ATOM_STR("\x3", "low");
```

Comment:
After some thoughts I realized we don't really want this pattern for defining atom strings:
- the compiler is going to de-duplicate strings regardless using this pattern or not
- we add lines in place of writing things such as `ATOM_STR("\x3", "low")` 
- maintaining code requires jumping back and forth, while inline atom strings are "local"

Reply:
I see you point completely, and the comments about the macros as well. Originally I wrote this without either, but thought the macros might improve readability, which led to defining the static const char strings, but as you point out jumping back and forth is worse for readability.